dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -734,35 +734,37 @@ void ExchangeReceiverBase<RPCContext>::connectionDone(
     const String & local_err_msg,
     const LoggerPtr & log)
 {
-    Int32 copy_live_conn = -1;
+    // The whole function should be protected by the lock,
+    // because once live_connections is equal to 0, waitAllConnectionDone
+    // function will return and the ExchangeReceiver may be destructed
+    // before we leave this function.
```

Comment:
Won't `thread_manager->wait();` blocks until `connectionDone` exits?

https://github.com/pingcap/tiflash/blob/70ea5e611e2b0e3672ca589006970b0b82621e90/dbms/src/Flash/Mpp/ExchangeReceiver.cpp#L347-L354

Reply:
> Won't `thread_manager->wait();` blocks until `connectionDone` exits?
> 
> https://github.com/pingcap/tiflash/blob/70ea5e611e2b0e3672ca589006970b0b82621e90/dbms/src/Flash/Mpp/ExchangeReceiver.cpp#L347-L354

No, new thread that reads data for local tunnel is no more need after this [pr](https://github.com/pingcap/tiflash/pull/6463)